include/constants/form_change_types.h

```diff
@@ -116,4 +116,7 @@
 // param2: multichoice list (starting at 0).
 #define FORM_CHANGE_ITEM_USE_MULTICHOICE        19
 
+// Form change that happens when Shaymin is frozen
+#define FORM_CHANGE_FROZEN        20
```

Comment:
Instead of it being only for freezing, I would recommend making this to be based on `STATUS1` and have the freezing and frostbite statuses as parameters.